core/chains/evm/txmgr/stuck_tx_detector.go

```diff
@@ -153,11 +153,28 @@ func (d *stuckTxDetector) FindUnconfirmedTxWithLowestNonce(ctx context.Context,
 		}
 	}
 
-	// Build list of potentially stuck tx but exclude any that are already marked for purge
+	// Build list of potentially stuck tx but exclude any that are already marked for purge or have non-broadcasted attempts
 	var stuckTxs []Tx
 	for _, tx := range lowestNonceTxMap {
-		// Attempts are loaded newest to oldest so one marked for purge will always be first
-		if len(tx.TxAttempts) > 0 && !tx.TxAttempts[0].IsPurgeAttempt {
+		if len(tx.TxAttempts) == 0 {
+			d.lggr.AssumptionViolationw("encountered an unconfirmed transaction without an attempt", "tx", tx)
+			continue
+		}
+		// Check the transaction's attempts in case any are already marked for purge or if any are not broadcasted
+		// We can only have one non-broadcasted attempt for a transaction at a time
+		// Skip purge detection until all attempts are broadcasted to avoid conflicts with the purge attempt
+		var foundPurgeAttempt, foundNonBroadcastAttempt bool
+		for _, attempt := range tx.TxAttempts {
+			if attempt.IsPurgeAttempt {
+				foundPurgeAttempt = true
+				break
```

Comment:
should we break here ?

Reply:
I think so. We only have to find 1 purge attempt in the tx to determine that it has already been marked for purge. So we can break out and skip adding it to the potential stuck tx list